### PR TITLE
posix: Support USB 3.2

### PR DESCRIPTION
**Pull request by Qwinci**

### CODE REVIEW

**Review by qookei — APPROVED** [review]

File: posix/subsystem/src/subsystem/usb/root-hub.hpp [hunk 1]
```diff
@@ -5,6 +5,26 @@
 
 namespace usb_subsystem::root_hub {
 
+constexpr std::array<uint8_t, 18> descUsb3_2 = {
+	0x12, /* bLength */
+	0x01, /* bDescriptorType */
+	0x20, 0x03, /* bcdUsb */
```
Comment (no92) [inline posix/subsystem/src/subsystem/usb/root-hub.hpp:11]: AFAICT Linux doesn't have a separate descriptor for USB 3.2; if we were to follow Linux, we would just use the 3.1 descriptor instead.

File: posix/subsystem/src/subsystem/usb/usb.cpp [hunk 2]
```diff
@@ -118,6 +118,11 @@ async::detached bindController(mbus::Entity entity, mbus::Properties properties,
 					device->descriptors.insert(device->descriptors.end(), root_hub::descUsb3_1.begin(), root_hub::descUsb3_1.end());
 					device->descriptors.insert(device->descriptors.end(), root_hub::descSuperSpeed.begin(), root_hub::descSuperSpeed.end());
 					break;
+				case 0x20:
+					device->speed = "20000";
+					device->descriptors.insert(device->descriptors.end(), root_hub::descUsb3_2.begin(), root_hub::descUsb3_2.end());
```
Comment (no92) [inline posix/subsystem/src/subsystem/usb/usb.cpp:123]: See above.

**Review by no92 — COMMENTED** [review]
LGTM but the question is whether we follow Linux or report USB 3.2 here. I don't think that really matters for anything, but would be nice to have a policy for.

**Comment by no92** [comment]
Implemented by #648.

**Pull request closed without merging**